src/main.cpp

```diff
@@ -5654,24 +5654,26 @@ bool SendMessages(CNode* pto)
         //
         // Message: getdata (non-blocks)
         //
-        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow)
+        if(!IsInitialBlockDownload())
```

Comment:
Yes, you're right.  I made the change and tested it out.

I pushed it, if you want to take a look before I squash.



On 14/12/2015 11:56 AM, Suhas Daftuar wrote:
>
> In src/main.cpp
> <https://github.com/bitcoin/bitcoin/pull/7164#discussion_r47548846>:
>
> > @@ -5654,24 +5654,26 @@ bool SendMessages(CNode* pto)
> >          //
> >          // Message: getdata (non-blocks)
> >          //
> > -        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow)
> > +        if(!IsInitialBlockDownload())
>
> I think the guard on line 4566 is sufficient, from my reading of the
> code and a quick test. Is there a case I'm missing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/7164/files#r47548846>.
>